corehq/apps/app_manager/fields.py

```diff
@@ -116,3 +137,387 @@ def get_app_sources(domain):
+                        form['app_does_not_exist'] = True
+                        form['possibilities'] = possibilities[form['xmlns']]
+                        if form['possibilities']:
+                            form['duplicate'] = True
+                    else:
+                        if app.domain != self.domain:
+                            logging.error("submission tagged with app from wrong domain: %s" % app.get_id)
+                        else:
+                            if app.copy_of:
+                                try:
+                                    app = app_cache[app.copy_of]
+                                    form['app_copy'] = {'id': app.get_id, 'name': app.name}
+                                except KeyError:
+                                    form['app_copy'] = {'id': app.copy_of, 'name': '?'}
+                            if app.is_deleted():
+                                form['app_deleted'] = {'id': app.get_id}
+                            try:
+                                app_forms = app.get_xmlns_map()[form['xmlns']]
+                            except AttributeError:
+                                # it's a remote app
+                                app_forms = None
+                                form['has_app'] = True
+                            if app_forms:
+                                app_form = app_forms[0]
+                                if app_form.doc_type == 'UserRegistrationForm':
+                                    form['is_user_registration'] = True
+                                else:
+                                    app_module = app_form.get_module()
+                                    form['module'] = app_module
+                                    form['form'] = app_form
+                                form['show_xmlns'] = False
+
+                            if not form.get('app_copy') and not form.get('app_deleted'):
+                                form['no_suggestions'] = True
+                    if app:
+                        form['app'] = {'id': app.get_id, 'name': app.name, 'langs': app.langs}
+                else:
+                    form['possibilities'] = possibilities[form['xmlns']]
+                    if form['possibilities']:
+                        form['duplicate'] = True
+                    else:
+                        form['no_suggestions'] = True
+        return unknown_forms
+
+    @staticmethod
+    def _sorkey_form(form):
+        app_id = form['app']['id']
+        if form.get('has_app', False):
+            order = 0 if not form.get('app_deleted') else 1
+            app_name = form['app']['name']
+            module = form.get('module')
+            if module:
+                # module is sometimes wrapped json, sometimes a dict!
+                module_id = module['id'] if 'id' in module else module.id
+            else:
+                module_id = -1 if form.get('is_user_registration') else 1000
```

Comment:
I actually grabbed this code from the old exports view that's going to be blown away. @dannyroberts was the original author https://github.com/dimagi/commcare-hq/blame/master/corehq/apps/reports/standard/export.py#L250

it just looks like two unlikely module numbers were chosen as a module id, and that it's very unlikely (unless there was something wrong with the app, or maybe if it was remote) that the `1000` would be used